File: Algoritimos 2/Lista 2/Exercicio8.c
#include <stdio.h>
int f(int v[], int a, int u[], int p, int l, int d);

void main() {
    int v[10] = {0, 0, 10, 2, 3, 4, 1};
    int u[10] = {10, 2, 3, 4, 1};
    printf("valor: %d\n", f(v, 2, u, 0, 4, 0));
}

int f(int v[], int a, int u[], int p, int l, int d) {
    if (a+d == a+l) return l + 1;
    if (v[a+d] != u[p+d]) return a;
    else return f(v, a, u, p, l, d+1);
}

File: Paradigmas/1° Semana/Exercicio8.java
import java.util.Scanner;
import java.lang.Float;

public class Exercicio8 {
    public static void main (String[] args) {
        int fatCount;
        float currentHeight, currentSize;
        Scanner input = new Scanner (System.in);
        
        fatCount = -1;

        do {
            System.out.print ("Digite o valor da altura: ");
            currentHeight = input.nextFloat ();

            System.out.print ("Digite o valor do peso: ");
            currentSize = input.nextFloat ();

            if (Float.compare ((currentSize / (currentHeight * currentHeight)), 25.0f) <= 0) 
                fatCount++;
        } while (currentHeight != -1 && currentSize != -1);

        System.out.printf ("A quantidade de pessoas acima do peso é: %d\n", fatCount);
    }
}